pmd-java/src/main/java/net/sourceforge/pmd/lang/java/rule/design/CouplingBetweenObjectsRule.java

```diff
@@ -145,11 +114,12 @@ private void checkVariableType(Node nameNode, String variableType) {
 
         // if the field is of any type other than the class type
         // increment the count
-        ClassScope clzScope = ((JavaNode) nameNode).getScope().getEnclosingScope(ClassScope.class);
-        if (!clzScope.getClassName().equals(variableType) && !this.filterTypes(variableType)
-                && !this.typesFoundSoFar.contains(variableType)) {
+        ClassScope clzScope = typeNode.getScope().getEnclosingScope(ClassScope.class);
+        String typeName = TypePrettyPrint.prettyPrint(typeNode.getTypeMirror());
```

Comment:
I don't think it's a good idea to couple a rule implementation to the pretty printer. We could let the `typesFoundSoFar` contain `JTypeMirror` instances instead of strings